src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+                                    records,
+                                    valid,
+                                    error))
+    return FALSE;
+  *valid = txt_v2_handler_checks (lan_data,
+                                  swbf,
+                                  records);
+  return TRUE;
+}
+
+static ServiceCheck
+get_txt_handler (const gchar *txt_version)
+{
+  guint64 v;
+  const gchar *end = NULL;
+
+  errno = 0;
+  v = g_ascii_strtoull (txt_version, (gchar **)&end, 10);
+
+  if (errno != 0 || end == NULL)
+    return NULL;
+
+  if (v == 1 && *end == '\0')
+    return txt_v1_handler;
+  if (v == 2 && *end == '\0')
+    return txt_v2_handler;
+
+  return NULL;
+}
+
+// Puts services with branch file signatures in front of services
+// without them.  Puts services with newer branch file timestamps in
+// front of services with older ones.
+static gint
+g_compare_func_swbf_by_timestamp (gconstpointer swbf1_ptr_ptr,
+                                  gconstpointer swbf2_ptr_ptr)
+{
+  EosServiceWithBranchFile *swbf1 = *((EosServiceWithBranchFile **)swbf1_ptr_ptr);
+  EosServiceWithBranchFile *swbf2 = *((EosServiceWithBranchFile **)swbf2_ptr_ptr);
+  gboolean with_signature1 = swbf1->declared_sha512sum == NULL;
+  gboolean with_signature2 = swbf2->declared_sha512sum == NULL;
+
+  if ((with_signature1 ^ with_signature2) == 0)
+    {
+      // either both have signatures or neither
+      GDateTime *timestamp1 = swbf1->declared_download_time ?
+        swbf1->declared_download_time :
+        swbf1->branch_file->download_time;
+      GDateTime *timestamp2 = swbf2->declared_download_time ?
+        swbf2->declared_download_time :
+        swbf2->branch_file->download_time;
+
+      return g_date_time_compare (timestamp2, timestamp1);
+    }
+  // one has signature, the other not
+  if (with_signature1)
+    return -1;
+  return 1;
+}
+
+static gboolean
+filter_services (LanData *lan_data,
+                 GPtrArray *found_services,
+                 GPtrArray **out_valid_services,
+                 GError **error)
+{
+  guint idx;
+  g_autoptr(GPtrArray) valid_services = object_array_new ();
+
+  for (idx = 0; idx < found_services->len; ++idx)
+    {
+      gpointer service_ptr = g_ptr_array_index (found_services, idx);
+      EosAvahiService *service = EOS_AVAHI_SERVICE (service_ptr);
+      TxtRecordError txt_error = TXT_RECORD_OK;
+      const gchar *txt_version = get_unique_txt_record (service->txt,
+                                                        "eos_txt_version",
+                                                        &txt_error);
+      ServiceCheck checker;
+      gboolean valid = FALSE;
+      g_autoptr(EosServiceWithBranchFile) swbf = NULL;
+
+      if (txt_version == NULL)
+        {
+          // TODO: message
+          continue;
+        }
+      checker = get_txt_handler (txt_version);
+      if (checker == NULL)
+        {
+          // TODO: message
+          continue;
```

Comment:
Just added something. Basically complaining about either the lack of the version number in txt records or an unknown version.